activerecord/lib/active_record/fixtures.rb

```diff
@@ -883,7 +883,11 @@ def set_fixture_class(class_names = {})
       def fixtures(*fixture_set_names)
         if fixture_set_names.first == :all
           fixture_set_names = Dir["#{fixture_path}/{**,*}/*.{yml}"]
-          fixture_set_names.map! { |f| f[(fixture_path.to_s.size + 1)..-5] }
+          fixture_path_globs = Dir[fixture_path].uniq.compact.sort_by!(&:length).reverse!
```

Comment:
maybe rely on the order the filesystem returns them ?
also why the uniq / compact ?